param_persist/agents/base.py

```diff
@@ -48,3 +52,106 @@ def update(self, instance, **kwargs):
         An abstract function to update a persisted param class.
         """
         raise NotImplementedError('The "update" function must be overridden in the agent child class.')
+
+    def get_serialized_param(self, instance):
+        """
+        Get the serialized parameter data.
+
+        Args:
+            instance: The instance to serialize
+
+        Returns:
+            Json serialization of the parameterized instance.
+        """
+        return json.loads(JSONSerialization.serialize_parameters(instance))
+
+    @staticmethod
+    def get_param_names(param_object):
+        """
+        Return list of all the names in param.
+        """
+        param_items = param_object.param.get_param_values()
+        param_names = list()
+        for item in param_items:
```

Comment:
Just had a thought on this: doesn't the ``keys()`` method on dictionaries return a list of the keys? If so, you could simplify this to:
```
param_names = param_items.keys()
```

Reply:
That isn't a dictionary.